chiron/mcmc.py

```diff
@@ -112,103 +170,802 @@ def run(
             thermodynamic_state, ThermodynamicState
         ), f"Thermodynamic state must be ThermodynamicState, not {type(thermodynamic_state)}"
 
-        self.integrator.run(
+        updated_sampler_state, updated_nbr_list = self.integrator.run(
             thermodynamic_state=thermodynamic_state,
             sampler_state=sampler_state,
             n_steps=self.nr_of_moves,
+            nbr_list=nbr_list,
         )
 
         if self.save_traj_in_memory:
             self.traj.append(self.integrator.traj)
             self.integrator.traj = []
 
+        self._move_iteration += 1
+
+        # The thermodynamic_state will not change for the langevin move
+        return updated_sampler_state, thermodynamic_state, updated_nbr_list
+
 
 class MCMove(MCMCMove):
     def __init__(
-        self, nr_of_moves: int, reporter: Optional[_SimulationReporter]
+        self,
+        nr_of_moves: int,
+        reporter: Optional[_SimulationReporter],
+        report_frequency: int = 1,
+        update_stepsize: bool = False,
+        update_stepsize_frequency: int = 100,
+        method: str = "metropolis",
     ) -> None:
-        super().__init__(nr_of_moves, reporter=reporter)
+        """
+        Initialize the move.
 
-    def apply_move(self):
+        Parameters
+        ----------
+        nr_of_moves
+            Number of moves to be applied in each call to update.
+        reporter
+            Reporter object for saving the simulation step data.
+        report_frequency
+            Frequency of saving the simulation data.
+        update_stepsize
+            Whether to update the "stepsize" of the move. Stepsize is a generic term for the key move parameters.
+            For example, for a simple displacement move this would be the displacement_sigma.
+        update_stepsize_frequency
+            Frequency of updating the stepsize of the move.
+        method
```

Comment:
We should be more explicit that this is the `acceptance_method`, and might consider encapsulating all behavior associated with Monte Carlo acceptance into a single class (rather than a string) so we can implement Metropolis-Hastings, Barker, and others.